### ISSUE TITLE
WashingPostCollection does not work properly

### ISSUE THREAD

**Issue by Peilin-Yang** [castorini/Anserini#375]
When I run to index the WashingtonPost collection v2: `~/Anserini/target/appassembler/bin/IndexCollection -collection WashingtonPostCollection -input /scratch2/more_collections/WashingtonPost.v2/WashingtonPost.v2/data/ -generator JsoupGenerator -index lucene-index.wash18.pos+docvectors -threads 44 -storePositions -storeDocvectors -optimize &>log.wash18.pos+docvectors`

The error occurs:
```
java.lang.NullPointerException
        at io.anserini.index.IndexCollection$IndexerThread.run(IndexCollection.java:198) [anserini-0.1.1-SNAPSHOT.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
2018-08-02 10:24:33,729 WARN  [main] index.IndexCollection (IndexCollection.java:365) - Unexpected difference between number of indexed documents and index maxDoc.
2018-08-02 10:24:33,730 INFO  [main] index.IndexCollection (IndexCollection.java:368) - # Final Counter Values
2018-08-02 10:24:33,730 INFO  [main] index.IndexCollection (IndexCollection.java:369) - indexed:                0
2018-08-02 10:24:33,730 INFO  [main] index.IndexCollection (IndexCollection.java:370) - empty:                 84
2018-08-02 10:24:33,730 INFO  [main] index.IndexCollection (IndexCollection.java:371) - unindexed:             84
2018-08-02 10:24:33,730 INFO  [main] index.IndexCollection (IndexCollection.java:372) - unindexable:            0
2018-08-02 10:24:33,731 INFO  [main] index.IndexCollection (IndexCollection.java:373) - skipped:                3
2018-08-02 10:24:33,731 INFO  [main] index.IndexCollection (IndexCollection.java:374) - errors:                 0
2018-08-02 10:24:33,736 INFO  [main] index.IndexCollection (IndexCollection.java:377) - Total 30,745 documents indexed in 00:00:24
```
And the indexer quits unexpectedly

**Comment by borislin** [comment]
I think we need to add the `.jl` extension to the `allowedFileSuffix` in `WashingtonPostCollection`.

**Comment by Kytabyte** [comment]
@Peilin-Yang I addressed this exception results from the `ContentObj` in [this line](https://github.com/castorini/Anserini/blob/8de9fc046fcb188957c171190be927a2317d5430/src/main/java/io/anserini/collection/WashingtonPostCollection.java#L118) may be `null`.

After handling the Null exception, the program works fine. However, I noticed that the number of `skipped files` is quite large compared to my original design, which may affect the correctness of our results. I suspect there're some other problems in our code. I'll address it and do the pull request together

**Comment by borislin** [comment]
@Kytabyte I think this [line](https://github.com/castorini/Anserini/blob/8de9fc046fcb188957c171190be927a2317d5430/src/main/java/io/anserini/collection/WashingtonPostCollection.java#L144) is redundant.

**Comment by Kytabyte** [comment]
The reason for this issue is because the `content` may not only be a `String`, but possibly `List<String>` or `Map<String, String>`. I addressed a dirty way to solve this problem just to get the indexer working as soon as possible. I will raise a PR and we can discuss a neater way to do this.

**Comment by Peilin-Yang** [comment]
@Kytabyte 
I think there are still two problems:

1. I think we should include _title_ and _fullcaption_ field.
2. When concatenating paragraphs we should explicitly set a splitter, e.g. `\n` or ` `. The default Lucene Analyzer will not split the terms by `.`

**Comment by Peilin-Yang** [comment]
Related to #383, you should also update the unit test and make sure it actually works

**Comment by Kytabyte** [comment]
> Related to #383, you should also update the unit test and make sure it actually works

@Peilin-Yang Has this issue already been solved?

**Comment by Peilin-Yang** [comment]
@Kytabyte I just merged the PR, you should be able to send PR on top of it.

**Comment by Peilin-Yang** [comment]
Merged #390
